### ISSUE TITLE
Code coverage is not working

### ISSUE THREAD

**Issue by AntonRomanchuk** [gematik/ref-OpenHealthCardKit#14]
## Problem
When I add your dependency via SPM, code coverage stops working. For clarification, I created new project with only this dependency - the same result. 

My SPM package file:
```
// swift-tools-version: 5.7.1
import PackageDescription

let package = Package(
    name: "SomePackage",
    platforms: [
        .iOS(.v15)
    ],
    products: [
        .library(
            name: "SomePackage",
            targets: ["SomePackage"]),
    ],
    dependencies: [
        .package(url: "https://github.com/gematik/ref-OpenHealthCardKit", from: "5.3.0"),
    ],
    targets: [
        .target(
            name: "SomePackage",
            dependencies: [
                .product(name: "HealthCardControl", package: "ref-OpenHealthCardKit"),
                .product(name: "NFCCardReaderProvider", package: "ref-OpenHealthCardKit"),
                .product(name: "HealthCardAccess", package: "ref-OpenHealthCardKit"),
                .product(name: "CardReaderProviderApi", package: "ref-OpenHealthCardKit"),
                .product(name: "Helper", package: "ref-OpenHealthCardKit"),
            ],
            cSettings: [ .unsafeFlags(["-w"]) ],
            swiftSettings: [ .unsafeFlags(["-suppress-warnings"]) ]
        ),
    ]
)
```

Additionally, I'm getting this message for every test run on Xcode Cloud:
> Failed to merge raw profiles in directory /Volumes/workspace/DerivedData/Build/ProfileData/86DCE839-AA1A-4025-97FB-6B816DCA956E to destination /Volumes/workspace/DerivedData/Build/ProfileData/86DCE839-AA1A-4025-97FB-6B816DCA956E/Coverage.profdata: Aggregation tool '/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/llvm-profdata' failed with exit code 1: warning: /Volumes/workspace/DerivedData/Build/ProfileData/86DCE839-AA1A-4025-97FB-6B816DCA956E/B70CE6AE-BC56-4151-AD4E-E887E802B1EB-3219.profraw: invalid instrumentation profile data (file header is corrupt) error: no profile can be merged

## Expected Behavior
Code coverage works as before 

## Actual Behavior
Code coverage is not working

## Steps to Reproduce the Problem

  1. Add this dependency via SPM
  2. Add some tests
  3. Run tests and check code coverage

## Specifications

  - Framework version: 5.3.0
  - Platform: iOS
  - Xcode versions: 15+
  - Subsystem: iOS 15-17

**Comment by AntonRomanchuk** [comment]
@sigabrtz any updates, plans?

**Comment by mfiebig** [comment]
@AntonRomanchuk I did some research and found the main issue being the OpenSSL wrapper. This one ist distributed with a binary, I guess that has to match at least to some extend to get the coverage working again. So if you change your dependencies to just include

.package(url: "https://github.com/gematik/OpenSSL-Swift", from: "4.1.0"),

instead of OHCKit, it will fail with the same error. We'll schedule a fresh build with Xcode 15.2 for that library. I have no permanent solution though (yet).

**Comment by sigabrtz** [comment]
@AntonRomanchuk We released a new Version 5.7.0 that depends on OpenSSL 4.2.0 (which itself contains a library built with XCode 15.2). 

With that the unwanted behavior seems to be fixed.

**Comment by AntonRomanchuk** [comment]
@sigabrtz it is working again, thank you!
